source/common/logger/ur_sinks.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+namespace sink
+{
+
+class Sink
+{
+public:
+    void log(const char *level, const char *format, va_list vlist) {
+        const int buf_size = 8192;
+        char buffer[buf_size];
+        unsigned position = 0;
+        
+        int ret = print_to_buffer(buffer, buf_size, "[%s]:", level);
+        if (ret > 0) {
+            position += ret;
+        }
+        
+        ret = std::vsnprintf(&buffer[position], buf_size, format, vlist);
+        if (ret < 0) {
+            printf("Call to vsnprintf failed!");
+        }
+
+	    ret = std::fputs(buffer, stdout);
+        if (ret < 0) {
+            printf("Call to fputs failed!");
+        }
+        std::fputs("\n", stdout);
+    }
+
+private:
+    int print_to_buffer(char *buffer, const int buf_size, const char *format, ...) {
```

Comment:
nit: this logic could be implemented as a standalone (global) function that returns std::string. This way you could use this formatting not only for logging/printing but also for creating messages (e.g. for exceptions).

Something like:
```
std::string format_to_str(const char *format, ...) {
    char buffer[8096];
    const int buf_size = 8096;

    va_list vlist;

    va_start(vlist, format);
    int ret = std::vsnprintf(buffer, buf_size, format, vlist);
    va_end(vlist);
    if (ret < 0) {
        throw std::runtime_error("vsnprintf failed");
    }

    return std::string(buffer);
}
```